cli/kata-check.go

```diff
@@ -388,3 +398,41 @@ func genericKvmIsUsable() error {
 
 	return nil
 }
+
+// genericCheckKVMExtension allows to query about the specifc kvm extensions
+// nolint: unused, deadcode
+func genericCheckKVMExtensions(extensions map[string]kvmExtension) (map[string]int, error) {
+	results := make(map[string]int)
+
+	flags := syscall.O_RDWR | syscall.O_CLOEXEC
+	kvm, err := syscall.Open(kvmDevice, flags, 0)
+	if err != nil {
+		return results, err
+	}
+	defer syscall.Close(kvm)
+
+	for name, extension := range extensions {
+		fields := logrus.Fields{
+			"type":        "kvm extension",
+			"name":        name,
+			"description": extension.desc,
+			"id":          extension.id,
+		}
+
+		ret, _, errno := syscall.Syscall(syscall.SYS_IOCTL,
+			uintptr(kvm),
+			uintptr(C.ioctl_KVM_CHECK_EXTENSION),
+			uintptr(extension.id))
+
+		results[name] = int(ret)
+
+		if errno != 0 || ret <= 0 {
```

Comment:
Does ARM syscall calling convention return 0 upon error? On x86_64, ret is either 0 or -1 so it seems this always fails.

Reply:
Hi~ @bergwolf 
my neglect. ARM will use return value to illustrate some hardware feature, just like this cap `KVM_CAP_ARM_VM_IPA_SIZE` , it will return 0 upon error, and specific IPA limit this host supports.
I'm referring the kvm extension check in qemu source code, it should be applied to all archs.
```
int kvm_check_extension(KVMState *s, unsigned int extension)
{
    int ret;

    ret = kvm_ioctl(s, KVM_CHECK_EXTENSION, extension);
    if (ret < 0) {
        ret = 0;
    }

    return ret;
}
```
So here, I think I should only return < 0 upon error. ;)

Reply:
I see. Thanks for the explanation. It's interesting to know that kvm ioctl can return zero upon failure.

Reply:
@Pennyzct -- if it isn't completely clear to someone as smart as @bergwolf, I think it warrants a comment to clarify the behavior in the code.  WDYT?

Reply:
can't agree more.🤭, I will add detailed comments asap.